### PR TITLE
Doc: document generate-makefile.ci

### PR DESCRIPTION
**Pull request by bergzand**
### Contribution description

This documents the `generate-makefile.ci` target

### Testing procedure

Read the added section to the advanced build system tricks

### Issues/PRs references

None

### CODE REVIEW

**Review by benpicco — APPROVED** [review]

File: doc/doxygen/src/advanced-build-system-tricks.md [hunk 1]
```diff
@@ -72,3 +72,17 @@ To do a repository wide analysis, you can use the script
...
+Most applications and tests include a `Makefile.ci` to indicate which boards can
+not compile the application or test. The content for these files can be
+generated via the script in
+~~~~~~~~~~~~~~~~~~~
+make -C $APPLICATION_DIRECTORY generate-Makefile.ci
+~~~~~~~~~~~~~~~~~~~
+This will compile and link the application for every board available and record
+the result in the Makefile.ci. This requires the toolchain for every target to
+be available. The target supports using docker via the `BUILD_IN_DOCKER=1`
+variable.
```
Comment (maribu) [inline doc/doxygen/src/advanced-build-system-tricks.md:88]: This is actually default behavior, if I recall correctly. This also makes sense: I (before updating newlib) often had smaller binaries than the CI toolchain, resulting in the output of `Makefile.ci` often not being useful. With `BUILD_IN_DOCKER=1` as default, the binary sizes should be the same as in the CI.

Comment (bergzand) [reply doc/doxygen/src/advanced-build-system-tricks.md:88]: Default it doesn't use docker to build for the `create-Makefile.ci` target here. I agree that it sounds like a sane default, but currently it is not the default.

Comment (maribu) [reply doc/doxygen/src/advanced-build-system-tricks.md:88]: OK, sorry for the noise

**Comment by maribu** [comment]
bors merge

**Comment by silkeh** [comment]
Err... isn't the target `create-Makefile.ci`? :confused:

**Comment by maribu** [comment]
The implementation now was updated to match the doc

**Pull request merged**